yb-voyager/cmd/analyzeSchema.go

```diff
@@ -995,21 +1012,28 @@ func reportDeferrableConstraintCreateTable(createTableNode *pg_query.Node_Create
 		*/
 		if column.GetColumnDef() != nil {
 			constraints := column.GetColumnDef().GetConstraints()
+			colName := column.GetColumnDef().GetColname()
 			if constraints != nil {
 				isForeignConstraint := false
 				isDeferrable := false
-				for _, constraint := range constraints {
+				constraintSuffix := ""
+				for idx, constraint := range constraints {
 					if constraint.GetConstraint().Contype == pg_query.ConstrType_CONSTR_FOREIGN {
 						isForeignConstraint = true
 					} else if slices.Contains(deferrableConstraintsList, constraint.GetConstraint().Contype) {
+						//Getting the constraint's suffix before the DEFERRABLE clause as the clause is applicable to that constraint
+						constraintSuffix = getConstraintSuffix(constraints[idx-1].GetConstraint().Contype)
 						isDeferrable = true
 					}
 				}
 				if !isForeignConstraint && isDeferrable {
 					summaryMap["TABLE"].invalidCount[sqlStmtInfo.objName] = true
+					generatedConName := fmt.Sprintf("%s_%s%s", tableName, colName, constraintSuffix)
+					specifiedConstraintName := column.GetConstraint().GetConname()
+					conName := lo.Ternary(specifiedConstraintName == "", generatedConName, specifiedConstraintName)
 					reportCase(fpath, DEFERRABLE_CONSTRAINT_ISSUE, "https://github.com/yugabyte/yugabyte-db/issues/1709",
 						"Remove these constraints from the exported schema and make the necessary changes to the application before pointing it to target",
-						"TABLE", fullyQualifiedName, sqlStmtInfo.formattedStmt, UNSUPPORTED_FEATURES, DEFERRABLE_CONSTRAINT_DOC_LINK)
+						"TABLE", fmt.Sprintf("%s, constraint: (%s)", fullyQualifiedName, conName), sqlStmtInfo.formattedStmt, UNSUPPORTED_FEATURES, DEFERRABLE_CONSTRAINT_DOC_LINK)
```

Comment:
Just noting: This would be a great case for storing these in metadata, rather than "objectName".

Reply:
Yes right!